buffer/src/main/java/io/netty/buffer/PoolChunk.java

```diff
@@ -306,26 +306,31 @@ private long allocateRun(int normCapacity) {
      * @return index in memoryMap
      */
     private long allocateSubpage(int normCapacity) {
-        int d = maxOrder; // subpages are only be allocated from pages i.e., leaves
-        int id = allocateNode(d);
-        if (id < 0) {
-            return id;
-        }
+        // Obtain the head of the PoolSubPage pool that is owned by the PoolArena and synchronize on it.
+        // This is need as we may add it back and so alter the linked-list structure.
+        PoolSubpage<T> head = arena.findSubpagePoolHead(normCapacity);
```

Comment:
because we want to minimize the overhead as much as possible. With this it is possible to allocate out of different PoolSubPage pools concurrently ;)